### PR TITLE
Skynet AI

### PR DESCRIPTION
**Pull request by Kmc2000**
## About The Pull Request

Syndicate pilot training programs have had their budget significantly increased.
https://i.gyazo.com/08501de257cbc9ad51f774e77dbfa97a.mp4

AIs have now been trained to perform fleet based roles and maneuvers, making each ship class behave distinctly.

Prepare to die :))

fix https://github.com/BeeStation/NSV13/issues/346

## Changelog
:cl:
add: Syndicate AI ships now use Skynet and have group tactics.
/:cl:

### CODE REVIEW

File: nsv13/code/modules/overmap/weapons.dm [hunk 1]
```diff
@@ -210,34 +196,34 @@
...
-	if(!linked_areas.len && role != MAIN_OVERMAP) //AI ships and fighters don't have interiors
+	if(ai_controlled || !linked_areas.len && role != MAIN_OVERMAP) //AI ships and fighters don't have interiors
 		if(torpedoes <= 0)
 			if(ai_controlled)
 				addtimer(VARSET_CALLBACK(src, torpedoes, initial(src.torpedoes)), 60 SECONDS)
-			return
-		fire_projectile(/obj/item/projectile/guided_munition/torpedo, target, homing = TRUE, speed=1, explosive = TRUE)
+			return FALSE
 		torpedoes --
+		fire_projectile(/obj/item/projectile/guided_munition/torpedo, target, homing = TRUE, speed=1, explosive = TRUE)
```
Comment (Karmic-Skink) [inline nsv13/code/modules/overmap/weapons.dm:213]: Speed = 3 for fire torp

File: nsv13/code/modules/overmap/weapons.dm [hunk 2]
```diff
@@ -210,34 +196,34 @@
...
-	if(!linked_areas.len && role != MAIN_OVERMAP) //AI ships and fighters don't have interiors
+	if(ai_controlled || !linked_areas.len && role != MAIN_OVERMAP) //AI ships and fighters don't have interiors
 		if(missiles <= 0)
 			if(ai_controlled)
 				addtimer(VARSET_CALLBACK(src, missiles, initial(src.missiles)), 60 SECONDS)
-			return
-		fire_projectile(/obj/item/projectile/guided_munition/missile, target, homing = TRUE, speed=3, explosive = TRUE)
+			return FALSE
 		missiles --
+		fire_projectile(/obj/item/projectile/guided_munition/missile, target, homing = TRUE, speed=3, explosive = TRUE)
```
Comment (Karmic-Skink) [inline nsv13/code/modules/overmap/weapons.dm:226]: Speed = 1 for fire missile

File: nsv13/code/modules/overmap/overmap.dm [hunk 3]
```diff
@@ -104,8 +104,8 @@
 	var/obj/weapon_overlay/last_fired //Last weapon overlay that fired, so we can rotate guns independently
 	var/atom/last_target //Last thing we shot at, used to point the railgun at an enemy.
 
-	var/torpedoes = 15 //Prevent infinite torp spam
-	var/missiles = 0 //Nothing should start with missiles
+	var/torpedoes = 2 //Prevent infinite torp spam
+	var/missiles = 4 //Nothing should start with missiles
```
Comment (Karmic-Skink) [inline nsv13/code/modules/overmap/overmap.dm:108]: Might want to update that comment.

File: nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/gauss_gun.dm [hunk 4]
```diff
@@ -152,8 +153,9 @@
 	dir = WEST
 
 /obj/machinery/ship_weapon/gauss_gun/proc/onClick(atom/target)
-	if(pdc_mode)
+	if(pdc_mode && world.time >= last_pdc_fire+2 SECONDS)
```
Comment (Jalleo) [inline nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/gauss_gun.dm:156]: You may need a (world.time >= (last_pdc_fire + 2 Seconds))

File: nsv13/code/modules/overmap/overmap.dm (hunk 3 above)
Comment (Jalleo) [reply nsv13/code/modules/overmap/overmap.dm:108]: Agreeded

**Pull request merged**